### PR TITLE
ET-1649 Create a lambda stack

### PR DESCRIPTION
**Pull request by mjarrowsmith4**
## 🚀 Please check if the PR fulfills these requirements

* [x] Has the ticket number been referenced in the PR title?
* [x] have you added a description and rationale for the changes?
* [x] Have you written new tests for your changes to capture behaviours, rather than just to hit code coverage?
* [x] Have you updated documentation, if needed for this change?

## 📖 What kind of change does this PR introduce, and why?
Add a lambda stack and a function that updates metadata when a file is dropped in

## 🤔 What is the current behaviour?
S3 bucket has default metadata

## 💻 What is the new behaviour
S3 bucket has metadata edited on `PUT`

## 🧪 Testing Instructions
Make sure tests pass
Upload a html file to the test S3 bucket
Confirm it triggers a lambda on Upload
Confirm the file has the correct metadata `Content-Type=text/html;charset=utf-8`

### CODE REVIEW

File: functions/updateMetadata/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    };
+
+    try {
+      // Update the metadata of the uploaded object
+      await s3Client.send(new CopyObjectCommand(params));
+      console.log(`Updated metadata for object: ${objectKey}`);
+    } catch (error) {
+      console.error(`Error updating metadata: ${error}`);
+    }
+  }
```
Comment (mjarrowsmith4) [inline functions/updateMetadata/index.ts:31]: Works well with `.html`, just need to do the same with `.txt` files too

File: tests/updateMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+jest.spyOn(s3.S3Client.prototype, 'send').mockImplementation();
+const consoleLogMock = jest.spyOn(console, 'log').mockImplementation();
+const consoleErrorMock = jest.spyOn(console, 'error').mockImplementation();
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('handler', () => {
+  test('should update metadata for an .html file with missing metadata', async () => {
```
Comment (mjarrowsmith4) [inline tests/updateMetadata.ts:18]: Will need a test for .txt as well

File: functions/updateMetadata/index.ts (hunk 1 above)
Comment (LBird96) [reply functions/updateMetadata/index.ts]: Added text/plain metadata for .txt files 🎉

File: bin/app.ts [hunk 3]
```diff
@@ -48,3 +49,14 @@ S3.deploy();
 new SftpTransferStack(app, `${G.APP_NAME}-transfer`, {
   ...defaultProps,
 });
+
+// TODO: Lambda stack
```
Comment (SamualDJPayne) [inline bin/app.ts:53]: Should this TODO be here?

Comment (LBird96) [reply bin/app.ts]: nope

File: tests/updateMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+import handler from '../functions/updateMetadata';
+
+jest.mock('@aws-sdk/client-s3', () => ({
+  ...jest.requireActual('@aws-sdk/client-s3'),
+  CopyObjectCommand: jest.fn(),
+}));
+
+jest.spyOn(s3.S3Client.prototype, 'send').mockImplementation();
+const consoleLogMock = jest.spyOn(console, 'log').mockImplementation();
+const consoleErrorMock = jest.spyOn(console, 'error').mockImplementation();
```
Comment (sendelivery) [inline tests/updateMetadata.ts:11]: Asking out of curiosity, what was wrong with how the global `console` instance was mocked previously?

Comment (LBird96) [reply tests/updateMetadata.ts:11]: Every time I tried to run them, they failed - something along the lines of "Could not find global.console". Replacing with this seemed to make it work immediately :/

**Pull request merged**